scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -470,58 +546,68 @@ We can correct this by applying a "shrinkage" procedure, which will adjust large
+head(shrink_results)
 ```
 
-Now we can get the updated results and combine with the `rowData` from the SCE object to extract the gene symbol. 
+If you look at our `shrink_results` object, we see that the genes are labeled with the Ensembl gene identifiers, as those were the row names of the pseudo-bulked SCE we used as input to build our `DESeq2` object.
+Although some of us may have all of the identifiers memorized by heart, it can be useful to have a human readable symbol in our results.
+Before we save the results as a file, we will grab the gene symbols from the `rowData` of our original SCE object and add them as a new column.
 
-```{r}
-# create a data frame with the results 
-deseq_results <- shrink_results %>%
+```{r add gene symbol}
+# convert shrunken results to a tibble and add Ensembl id as a column
+deseq_results <- shrink_results |>
   tibble::as_tibble(rownames = "ensembl_id")
 
-# first look at the significant results 
-deseq_results_sig <- deseq_results %>%
-  dplyr::filter(padj <= 0.05)
-
-deseq_results_sig
-```
-
-
-```{r}
-# convert rowdata to dataframe 
-sce_rowdata_df <- rowData(tumor_myoblast_sce) %>%
+# convert rowdata to data frame 
+sce_rowdata_df <- rowData(tumor_myoblast_sce) |>
   # create a column with rownames stored as ensembl id to use for joining with deseq results
   tibble::as_tibble(rownames = "ensembl_id")
 
 # combine deseq results with rowdata by ensembl id 
-deseq_results <- deseq_results %>%
+deseq_results <- deseq_results |>
   dplyr::left_join(sce_rowdata_df, by = "ensembl_id")
 
 head(deseq_results)
 ```
 
-```{r}
-# save our results 
+We can save the new data frame that we have created with the Ensembl identifiers, gene symbols, and the `DESeq2` results as a tab separated (`tsv`) file.
+
+```{r save deseq}
+# save our results as tsv
 readr::write_tsv(deseq_results, deseq_output_file)
 ```
 
+The last thing that we will do is take a look at how many genes are significant.
+Here we will want to use the adjusted p-value, found in the `padj` column of the results, as this accounts for multiple test correction.
 
-### Exploring the identified differentially expressed genes 
+```{r significant reults}
+# first look at the significant results 
+deseq_results_sig <- deseq_results |>
+  # filter based on adjusted pvalue
+  dplyr::filter(padj <= 0.05)
+
+head(deseq_results_sig)
+```
 
-Let's also do some initial analysis to look at what types of genes are being identified and see if what is being identified as actually up/down regulated appear to correlate back to the single cell data. 
 
-The first plot we'll make is a volcano plot and we can label the significant genes with their gene symbols. 
+### Exploring the identified differentially expressed genes 
+
+Now that we have identified a set of genes that are DE in the tumor myoblasts between ARMS and ERMS subtypes, lets actually take a look at them and see if we can make some informative plots.
```

Comment:
I don't really like the use of "DE" here since I feel like it stands for "differential expression" but here it's used as "differentially expressed." So, potentially, is this just me? If not..

```suggestion
Now that we have identified a set of genes that are differentially expressed in the tumor myoblasts between ARMS and ERMS subtypes, lets actually take a look at them and see if we can make some informative plots.
```